### PR TITLE
Questions was done but My questions not finished yet

### PR DESCRIPTION
**Pull request by Nalog00**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -75,11 +76,15 @@ dependencies {
     implementation "androidx.navigation:navigation-ui-ktx:$nav_version"
     implementation "androidx.navigation:navigation-dynamic-features-fragment:$nav_version"
     androidTestImplementation "androidx.navigation:navigation-testing:$nav_version"
+    implementation "androidx.fragment:fragment-ktx:1.3.3"
```
Comment (texnopos) [inline app/build.gradle]: Mina lib kerek emes

File: app/build.gradle [hunk 2]
```diff
@@ -19,6 +19,7 @@ android {
         versionName "1.0"
 
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
```
Comment (texnopos) [inline app/build.gradle]: Minani alip taslaw kerek

File: app/src/main/java/uz/texnopos/paziylet/data/model/FirebaseHelper.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import android.content.Context
+import android.util.Log
+import com.google.firebase.firestore.FirebaseFirestore
+import org.koin.experimental.property.inject
+
+class FirebaseHelper(private val context: Context,private val db: FirebaseFirestore) {
+    fun getQuestionCategories(onSuccess: (list: List<QuestionCategories>)-> Unit,onFailure: (msg: String?)-> Unit){
+        db.collection("questionCategories").get()
+            .addOnSuccessListener {
+                if (it.documents.isNotEmpty()){
```
Comment (texnopos) [inline app/src/main/java/uz/texnopos/paziylet/data/model/FirebaseHelper.kt]: mina if kerek emes else penen qosip

File: app/src/main/java/uz/texnopos/paziylet/ui/questions/QuestionsFragment.kt [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import uz.texnopos.paziylet.ui.questions.viewPager1.ViewPager1Fragment
+import uz.texnopos.paziylet.ui.questions.viewPager2.ViewPager2Fragment
+
+class QuestionsFragment: Fragment(R.layout.fragment_questions) {
+    private lateinit var navController: NavController
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        navController = Navigation.findNavController(view)
+        val fragmentList = arrayListOf(
+            ViewPager1Fragment(),
```
Comment (texnopos) [inline app/src/main/java/uz/texnopos/paziylet/ui/questions/QuestionsFragment.kt]: Rename: QuestionCategoriesFragment and MyQuestionsFragment

File: app/src/main/java/uz/texnopos/paziylet/ui/questions/QuestionsFragment.kt [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            ViewPager2Fragment()
+        )
+
+        val adapter= ViewPagerAdapter(fragmentList,requireActivity().supportFragmentManager,lifecycle)
+        viewPager.adapter=adapter
+        val tabLayoutMediator =
+            TabLayoutMediator(tabs,viewPager
+            ) { tab, position ->
+                when (position) {
+                    0 -> tab.text = "Сораўлар"
```
Comment (texnopos) [inline app/src/main/java/uz/texnopos/paziylet/ui/questions/QuestionsFragment.kt]: extract to string file
name = questions
name = my_questions

File: app/src/main/java/uz/texnopos/paziylet/ui/questions/viewPager1/ViewPager1Adapter.kt [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.synthetic.main.item_view_pager1.view.*
+import uz.texnopos.paziylet.R
+import uz.texnopos.paziylet.data.model.QuestionCategories
+
+class ViewPager1Adapter: RecyclerView.Adapter<ViewPager1Adapter.ViewPager1ViewHolder>() {
```
Comment (texnopos) [inline app/src/main/java/uz/texnopos/paziylet/ui/questions/viewPager1/ViewPager1Adapter.kt]: rename to QuestionCategoriesAdapter

File: app/src/main/java/uz/texnopos/paziylet/ui/questions/viewPager1/ViewPager1Adapter.kt [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+    var models: List<QuestionCategories> = listOf()
+    set(value) {
+        field = value
+        notifyDataSetChanged()
+    }
+
+    inner class ViewPager1ViewHolder(itemView: View): RecyclerView.ViewHolder(itemView){
```
Comment (texnopos) [inline app/src/main/java/uz/texnopos/paziylet/ui/questions/viewPager1/ViewPager1Adapter.kt]: QuestionCategoriesViewHolder

File: app/src/main/java/uz/texnopos/paziylet/ui/questions/viewPager1/ViewPager1Adapter.kt [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+        fun populateModel(model: QuestionCategories){
+            itemView.tvCategoryName.text = model.name
+        }
+
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewPager1ViewHolder {
+        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.item_view_pager1,parent,false)
```
Comment (texnopos) [inline app/src/main/java/uz/texnopos/paziylet/ui/questions/viewPager1/ViewPager1Adapter.kt]: extensions tin ishinde parent.inflate dep jiberetugin funkciya jazilgan edigo

File: app/src/main/java/uz/texnopos/paziylet/ui/questions/viewPager1/ViewPager1Fragment.kt [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+import androidx.navigation.NavController
+import androidx.navigation.Navigation
+import kotlinx.android.synthetic.main.fragment_view_pager1.*
+import org.koin.android.ext.android.inject
+import org.koin.android.viewmodel.ext.android.viewModel
+import uz.texnopos.paziylet.R
+import uz.texnopos.paziylet.core.extentions.visibility
+import uz.texnopos.paziylet.di.ResourceState
+
+class ViewPager1Fragment: Fragment(R.layout.fragment_view_pager1) {
```
Comment (texnopos) [inline app/src/main/java/uz/texnopos/paziylet/ui/questions/viewPager1/ViewPager1Fragment.kt]: rename

File: app/src/main/java/uz/texnopos/paziylet/ui/questions/viewPager1/ViewPager1Fragment.kt [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+        private val viewModel: ViewPagerViewModel by viewModel()
+        private val adapter: ViewPager1Adapter by inject()
+        override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        recViewPager1.adapter = adapter
```
Comment (texnopos) [inline app/src/main/java/uz/texnopos/paziylet/ui/questions/viewPager1/ViewPager1Fragment.kt]: recViewPager1 -> rvCategories

File: app/src/main/java/uz/texnopos/paziylet/ui/questions/viewPager1/ViewPager1Fragment.kt [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+        viewModel.getAllQuestionCategories()
+            setUpObserver()
```
Comment (texnopos) [inline app/src/main/java/uz/texnopos/paziylet/ui/questions/viewPager1/ViewPager1Fragment.kt]: tabulyaciya ne qilip tur aldinda?

File: app/src/main/java/uz/texnopos/paziylet/ui/questions/viewPager1/ViewPager1Fragment.kt [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+    }
+    private fun setUpObserver(){
+        viewModel.questionCategories.observe(viewLifecycleOwner,{
+            when(it.status){
+                ResourceState.LOADING-> progressBarViewPager.visibility(true)
+                ResourceState.SUCCESS->{
+                    viewModel.questionCategories.observe(viewLifecycleOwner,{ i->
```
Comment (texnopos) [inline app/src/main/java/uz/texnopos/paziylet/ui/questions/viewPager1/ViewPager1Fragment.kt]: success dep juwap kelgende nege jane observe qilip atirsan?
 o'shir buni tek 
adapter.models = i.data!!
progressBarViewPager.visibility(false)
dep jaz

File: app/src/main/java/uz/texnopos/paziylet/ui/questions/viewPager1/ViewPagerViewModel.kt [hunk 13]
```diff
@@ -0,0 +1,27 @@
+package uz.texnopos.paziylet.ui.questions.viewPager1
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import uz.texnopos.paziylet.data.model.FirebaseHelper
+import uz.texnopos.paziylet.data.model.QuestionCategories
+import uz.texnopos.paziylet.di.Resource
+
+class ViewPagerViewModel(private val firebaseHelper: FirebaseHelper): ViewModel() {
```
Comment (texnopos) [inline app/src/main/java/uz/texnopos/paziylet/ui/questions/viewPager1/ViewPagerViewModel.kt]: rename it. QuestionCategoriesViewModel dep

File: app/src/main/res/drawable/ic_mechet1.xml [hunk 14]
```diff
@@ -0,0 +1,57 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (texnopos) [inline app/src/main/res/drawable/ic_mechet1.xml]: Mina fayldin ati nege bunday?

File: app/src/main/res/layout/activity_main.xml [hunk 15]
```diff
@@ -22,10 +21,10 @@
         android:layout_height="wrap_content"
         android:layout_alignParentBottom="true"
         android:layout_gravity="bottom"
-        android:background="#37B899"
-        app:itemIconTint="@color/white"
-        app:itemTextColor="@color/white"
-        app:labelVisibilityMode="labeled"
+        android:background="#FFFFFF"
```
Comment (texnopos) [inline app/src/main/res/layout/activity_main.xml:24]: @color/white dep jaz

File: app/src/main/res/layout/activity_main.xml [hunk 16]
```diff
@@ -22,10 +21,10 @@
...
+        android:elevation="5dp"
+        app:itemIconTint="#62e4c4"
```
Comment (texnopos) [inline app/src/main/res/layout/activity_main.xml:26]: mina renlerge at ber

File: app/src/main/res/layout/fragment_questions.xml [hunk 17]
```diff
@@ -0,0 +1,69 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+
```
Comment (texnopos) [inline app/src/main/res/layout/fragment_questions.xml:8]: bir line lishniy

File: app/src/main/res/layout/fragment_questions.xml [hunk 18]
```diff
@@ -0,0 +1,69 @@
...
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:id="@+id/appBarLayout"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="@color/colorAppBar"
+        tools:ignore="MissingConstraints">
```
Comment (texnopos) [inline app/src/main/res/layout/fragment_questions.xml:14]: MissingConstraints degen ne qilip tur? Buni oshir, viewlerge constraint qasiyetlerin berip shesh bul problemani

File: app/src/main/res/layout/fragment_questions.xml [hunk 19]
```diff
@@ -0,0 +1,69 @@
...
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginBottom="16dp"
+            android:background="@drawable/ic_nagis"
+            app:layout_constraintBottom_toTopOf="@id/tabs"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <ImageView
+            android:layout_width="110dp"
```
Comment (texnopos) [inline app/src/main/res/layout/fragment_questions.xml:38]: nege minagan dp berip qoyipsan? match parent ayamasa wrap emes

File: app/src/main/res/layout/fragment_view_pager1.xml [hunk 20]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (texnopos) [inline app/src/main/res/layout/fragment_view_pager1.xml]: rename this file to fragment_question_categories

File: app/src/main/res/layout/fragment_view_pager1.xml [hunk 21]
```diff
@@ -0,0 +1,44 @@
...
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:background="#edf1f4">
```
Comment (texnopos) [inline app/src/main/res/layout/fragment_view_pager1.xml]: colorga qos rendi

File: app/src/main/res/values/strings.xml [hunk 22]
```diff
@@ -1,4 +1,10 @@
 <resources>
     <string name="app_name">Paziylet</string>
     <string name="news">Jańalıqlar</string>
+    <string name="home">Бас бет</string>
+    <string name="favourite">Сайландылар</string>
+    <string name="questions">Сорау-Жуап</string>
```
Comment (texnopos) [inline app/src/main/res/values/strings.xml]: question_answer dep ozgert, value di de Сораў-жуўап dep o'zgert

File: app/src/main/res/values/strings.xml [hunk 23]
```diff
@@ -1,4 +1,10 @@
...
+    <string name="time_namaz">Намаз уакты</string>
```
Comment (texnopos) [inline app/src/main/res/values/strings.xml]: ўақты

File: app/src/main/res/values/strings.xml [hunk 24]
```diff
@@ -1,4 +1,10 @@
...
+    <string name="compass">Компас</string>
+    <string name="question_answer">Сораў жуўап</string>
```
Comment (texnopos) [inline app/src/main/res/values/strings.xml:8]: Joqarida soraw juwap degen bargo, ekewinin parqi ne?

File: app/src/main/res/values/themes.xml [hunk 25]
```diff
@@ -1,9 +1,9 @@
 <resources xmlns:tools="http://schemas.android.com/tools">
     <!-- Base application theme. -->
-    <style name="Theme.Paziylet" parent="Theme.MaterialComponents.DayNight.DarkActionBar">
+    <style name="Theme.Paziylet" parent="Theme.MaterialComponents.DayNight.NoActionBar">
         <!-- Primary brand color. -->
         <item name="colorPrimary">@color/purple_500</item>
-        <item name="colorPrimaryVariant">@color/purple_700</item>
+        <item name="colorPrimaryVariant">#289C81</item>
```
Comment (texnopos) [inline app/src/main/res/values/themes.xml:6]: purple_700 din irnina #289C81 di qoyajaqsanba onda purple_700 din value sin ozgertta

File: app/src/main/res/values/strings.xml (hunk 24 above)
Comment (Nalog00) [reply app/src/main/res/values/strings.xml:8]: Burewinde "-" bar burewinde joq

**Pull request merged**